### PR TITLE
Us119 display chargin clause

### PR DESCRIPTION
**Pull request by Kirosoft**
- Do Not Merge - Initial pull request pending matching front end changes and acceptance tests*

Review includes:

data_importer.py tool in the migrations directory which can import csv data directly into sql tables. This is used to setup initial import data.

### CODE REVIEW

File: migrations/setup_initial_data/data/mortgage_document/test_md.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+md_ref|data
+e-MD12344|{"description":"test setup charge clause","lender":{ "name":"a new lender","address":"no 1 reputable street","registration":"companyNum"}, "charge_clause": { "cre_code": "CRE001","description":"This is an example charge clause"}, "additional_provisions": [ { "additional_provision_code":"addp001", "description":"this is additional provision1"}, { "additional_provision_code":"addp002", "description":"this is additional provision2"}]}
+MD559|{"description":"test setup charge clause2","lender":{ "name":"a new lender2","address":"no 2 reputable street","registration":"companyNum"}, "charge_clause": { "cre_code": "CRE002","description":"This is an example charge clause"}, "additional_provisions": [ { "additional_provision_code":"addp003", "description":"this is additional provision 3"}, { "additional_provision_code":"addp004", "description":"this is additional provision 4"}]}
```
Comment (danielmarker) [inline migrations/setup_initial_data/data/mortgage_document/test_md.csv]: Are we happy putting in Non E-MD refs?

File: unit_tests/helper.py [hunk 2]
```diff
@@ -3,6 +3,16 @@ class DeedModelMock:
     deed = {"title_number": "DN100"}
 
 
+class MortgageDocMock:
+    md_ref = "MD12121"
```
Comment (danielmarker) [inline unit_tests/helper.py]: Again - are we happy this is not a E- Md?

File: migrations/versions/390d0c59875_.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 # revision identifiers, used by Alembic.
 revision = '390d0c59875'
-down_revision = '610e8a67d8'
+down_revision = 'cfe2e2f325'
```
Comment (tsangsimon) [inline migrations/versions/390d0c59875_.py:12]: I think this edited migration file is incorrectly defined. The down revision is the newest migration for the mortgage_document table. I think it should be the original. not sure how alembic will handle this

File: integration_tests/helper.py [hunk 4]
```diff
@@ -24,6 +26,17 @@ def setUpDB(self):
         db.create_all()
 
 
+def setUp_MortgageDocuments(self):
+    engine = create_engine('postgresql://vagrant:vagrant@localhost:5432/deed_api', convert_unicode=True)
```
Comment (tsangsimon) [inline integration_tests/helper.py]: May want to load these as environment variables

File: application/deed/deed-api.json [hunk 5]
```diff
@@ -157,6 +171,15 @@
                         },
                         "borrowers": {
                             "$ref": "#/definitions/Borrowers"
+                        },
+                        "charge_clauses": {
```
Comment (tsangsimon) [inline application/deed/deed-api.json]: Think this is a single element and will never be an array. please refactor the ChargeClauses definition

File: migrations/setup_initial_data/data/mortgage_document/test_md.csv [hunk 6]
```diff
@@ -0,0 +1,3 @@
+md_ref|data
+e-MD12344|{"description":"test setup charge clause","lender":{ "name":"a new lender","address":"no 1 reputable street","registration":"companyNum"}, "charge_clause": { "cre_code": "CRE001","description":"This is an example charge clause"}, "additional_provisions": [ { "additional_provision_code":"addp001", "description":"this is additional provision1"}, { "additional_provision_code":"addp002", "description":"this is additional provision2"}]}
```
Comment (tsangsimon) [inline migrations/setup_initial_data/data/mortgage_document/test_md.csv]: Can we have the real data in here before we merge this into develop

File: migrations/setup_initial_data/data_importer.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+def build_data(cols, row):
+    data = {}
+    for idx, col in enumerate(cols):
+        data[col] = row[idx]
+    return [data]
+
+
+def process_input_options():
+    parser = argparse.ArgumentParser(
+            prog='data_importer', description='utility to import all '
```
Comment (endamccormack) [inline migrations/setup_initial_data/data_importer.py]: This looks like wrong indentation... how did this pass flake8

File: integration_tests/helper.py (hunk 4 above)
Comment (endamccormack) [inline integration_tests/helper.py]: :+1:

**Comment by danielmarker** [comment]
:+1:

**Comment by endamccormack** [comment]
:+1:

**Pull request merged**